shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegateTest.mm

```diff
@@ -89,6 +95,21 @@ - (void)testWillTerminate {
   [self waitForExpectations:@[ expectation ] timeout:5.0];
   OCMVerify([plugin applicationWillTerminate:[UIApplication sharedApplication]]);
 }
+
+- (void)testReleasesPluginOnDealloc {
+  __weak id<FlutterApplicationLifeCycleDelegate> weakPlugin;
+  __weak FlutterPluginAppLifeCycleDelegate* weakDelegate;
+  @autoreleasepool {
+    FakePlugin* fakePlugin = [[FakePlugin alloc] init];
+    weakPlugin = fakePlugin;
+    FlutterPluginAppLifeCycleDelegate* delegate = [[FlutterPluginAppLifeCycleDelegate alloc] init];
+    [delegate addDelegate:fakePlugin];
+    weakDelegate = delegate;
+  }
+  XCTAssertNil(weakPlugin);
+  XCTAssertNil(weakDelegate);
```

Comment:
This fails on main due to the retain cycle:
https://github.com/flutter/engine/blob/aeff9b174c848c52a984b8a40503022c12644c77/shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegate.mm#L43-L47